src/grid.cpp

```diff
@@ -553,8 +553,11 @@ diagnostics_t grid::diagnostics(const diagnostics_t &diags) {
 						rc.gt[i] += dX[0] * G[iiig][gy_i] * dV * rho0;
 						rc.gt[i] -= dX[1] * G[iiig][gx_i] * dV * rho0;
 						const auto ekin = (pow(U[sx_i][iii], 2) + pow(U[sy_i][iii], 2) + pow(U[sz_i][iii], 2)) / 2.0 / U[rho_i][iii] * dV;
-						const auto eint = U[egas_i][iii] * dV - ekin;
+						auto eint = U[egas_i][iii] * dV - ekin;
 						const auto epot = 0.5 * U[pot_i][iii] * dV;
+						if (eint < de_switch2 * U[egas_i][iii]) {
```

Comment:
U[egas_i][iii] should be multiplied by dV too.
In addition, the condition should be less equal (<=) and not just smaller than (<).